src/shims/env.rs

```diff
@@ -104,11 +103,52 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         if let Some(old) = success {
             if let Some(var) = old {
-                this.memory_mut().deallocate(var, None, MiriMemoryKind::Env.into())?;
+                this.memory_mut()
+                    .deallocate(var, None, MiriMemoryKind::Env.into())?;
             }
             Ok(0)
         } else {
             Ok(-1)
         }
     }
+
+    fn getcwd(
+        &mut self,
+        buf_op: OpTy<'tcx, Tag>,
+        size_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, Scalar<Tag>> {
+        let this = self.eval_context_mut();
+
+        if !this.machine.communicate {
+            throw_unsup_format!("`getcwd` not available when isolation is enabled")
+        }
+
+        let tcx = &{ this.tcx.tcx };
+
+        let buf = this.force_ptr(this.read_scalar(buf_op)?.not_undef()?)?;
+        let size = this.read_scalar(size_op)?.to_usize(&*this.tcx)?;
```

Comment:
You have a `tcx` in scope, why not use that?

Reply:
Oh... sure, let me correct that.

Reply:
https://github.com/rust-lang/miri/pull/983/commits/60cf06a03fa06bc1c5e2981d4ad7f5496b06c62f